app/services/email_sender_service/email_sender.rb

```diff
@@ -0,0 +1,45 @@
+module EmailSenderService
+  class EmailSender
+    def call(address:, subject:, body:)
+      provider.call(
+        address: email_or_email_override(address),
+        subject: subject,
+        body: body
+      )
+    end
+
+  private
+
+    def config
+      EmailAlertAPI.config.email_service
+    end
```

Comment:
I'd consider switching this to be injected in when we create this class (as that's only done in one place). 

You might even (with a tweak to the naming format) be able to get it down to this:

```
def initialize(config) # or could have keyword args
  @email_override = config[:email_address_override]
  provider = "EmailSenderService::#{config.fetch(:provider).classify}"
  @provider = provider.constantize.new
rescue NameError
  raise "Email service provider #{provider} does not exist"
end
```

Reply:
so I've moved this to t'other PR (https://github.com/alphagov/email-alert-api/pull/215) - I've chosen not to do the `constantize` dance because I find it obscures the code a bit, mostly when I come back to it a few months later. It becomes hard to grep where classes are being used when tracing code through and debugging and for me, just provides that extra bit of mental overhead that the more specific individual instantiation of classes provides. Happy to chat about it though if anyone thinks `constantize` is better.